lib/puppet/type/constraint.rb

```diff
@@ -1,5 +1,5 @@
 module Puppet
-  newtype(:constraint) do
+  Puppet::Type.newtype(:constraint) do
```

Comment:
actually, we can get rid of the "module" bit, and its entire indentation level.

@ffrank what'd you say?